### ISSUE TITLE
[Enhancement] Add possibility to make multipick for Media Picker

### ISSUE THREAD

**Issue by Exis19ce** [xamarin/Essentials#1410]
Currently, we can pick only a single photo.

**Comment by mattleibow** [comment]
I wonder what the actual changes will be. Is this just a flag to the native implementation, or a whole new logic? 🤔

**Comment by oleksandrtretiak** [comment]
I have read in the documentation of PHPickerViewController that this controller provides multiple-picker, just need to add SelectionLimit for PHPickerConfiguration. Need to wait when @jamesmontemagno will upgrade UIImagePickerController.

**Comment by mattleibow** [comment]
I have a PR for the upgrade to `PHPickerConfiguration` #1475 

However, this might break VS2017, but I can't seem to even get VS2017 to connect to Apple to install certs.

**Comment by MatthewKapteyn** [comment]
@mattleibow @oleksandrtretiak 

given that james' CrossMedia MediaPlugin is being archived soon, really hoping this feature is added. The PHPicker library does make this pretty easy on iOS however support will be limited to iOS 14+. 

On Android you just have to insert the EXTRA_ALLOW_MULTIPLE to the intent. However some apps like the Samsung Gallery ignore it :(

If you want some sample code for PHPicker
https://github.com/MatthewKapteyn/MediaPlugin/blob/master/src/Media.Plugin/iOS/MediaPickerService.cs

**Comment by groege** [comment]
> On Android you just have to insert the EXTRA_ALLOW_MULTIPLE to the intent. However some apps like the Samsung Gallery ignore it :(

@MatthewKapteyn
Yes but there is always the option to install another gallery app to enable the multi picker - I'm not too familiar with Android (only the things I've needed so far) but is it possible to recommend a Gallery for Android? So we could set a requested app and when opening the gallery the store opens with a supplied recommendation?

**Comment by michaldivis** [comment]
Is this happening? I'd really appreciate the ability to pick multiple photos...

**Comment by MatthewKapteyn** [comment]
@groege 
Oops, sorry I missed this...a few months ago D:

My go to gallery app is the google photo gallery. It's pre-installed on most phones already including samsung.

Code snippet from my fork, I just ignore the default gallery because I got tired of complaints that multi picking wasn't working due to whatever default gallery app they had.

`
pickIntent.SetPackage("com.google.android.apps.photos");
pickIntent.PutExtra(Intent.ExtraAllowMultiple, true);
`

**Comment by konradzuse** [comment]
2 years since this was requested, should I just give up?

**Comment by MatthewKapteyn** [comment]
Still waiting here too but I understand the team has been pretty busy with MAUI. 
I've been hoping to discard my aging Media Plugin fork but it still continues to chug along.
